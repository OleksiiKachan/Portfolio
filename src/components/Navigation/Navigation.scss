.navigation {
  padding: 8px;
  height: 15%;

  background-color: #424242;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  transition: all 0.4s ease-in-out;

  position: relative;
  box-shadow: 3px 3px 15px 0px rgba(66, 66, 66, 0.6);
}

@media screen and (min-width: 865px) {
  .navigation_scrolled {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 2;
    opacity: 0.9;
    height: 10%;

    .navigation__logoWrapper {
      padding: 16px;
      width: 35%;
    }

    .navigation__logoImage {
      width: 40%;
    }
  }
}

.navigation__menuButton {
  width: 0;
}

.navigation__logoWrapper {
  width: 0;
  transition: all 0.4s ease-in-out;
}

.navigation__logoImage {
  width: 0;
  transition: all 0.4s ease-in-out;
}

.navigation__listWrapper {
  padding: 0 34px;
  width: 100%;
  transition: all 0.4s ease-in-out;

  display: flex;
  justify-content: center;
  align-items: center;
}

.navigation__list {
  list-style: none;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  width: 100%;
}

.navigation__listLink {
  display: inline-block;
  outline: none;
  color: #ddd;
  text-decoration: none;
  font-size: 14px;
  letter-spacing: 1.2px;
  font-weight: 500;
}

@media screen and (max-width: 864px) {
  .navigation__logoWrapper {
    padding: 0 8px;
    display: flex;
    justify-content: flex-start;
    align-items: center;
  }
  .navigation__logoWrapper {
    width: 25%;
  }

  .navigation__logoImage {
    content: url('https://res.cloudinary.com/oleksiikachan/image/upload/v1546142703/portfolio/images/logo/logo_square_light.svg');
    width: 45%;
  }

  .navigation__listWrapper {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: -1;
    background: #fff;
    opacity: 0;
    transition: display 0.5s ease-in-out;
  }

  .navigation__list {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
  }

  .navigation__listItem {
    display: block;
    float: none;
    width: 100%;
    text-align: right;
    margin-bottom: 10px;
  }

  .navigation__listItem:hover {
    opacity: 1;
  }

  .navigation__list li:nth-child(1) a {
    transition-delay: 0.2s;
  }

  .navigation__list li:nth-child(2) a {
    transition-delay: 0.3s;
  }

  .navigation__list li:nth-child(3) a {
    transition-delay: 0.4s;
  }

  .navigation__list li:nth-child(4) a {
    transition-delay: 0.5s;
  }

  .navigation__list li:nth-child(5) a {
    transition-delay: 0.6s;
  }

  .navigation__listLink {
    padding: 10px 24px;
    opacity: 0;
    color: #424242;
    font-size: 14px;
    font-weight: 600;
    letter-spacing: 1.2px;
    transform: translateX(-20px);
    transition: all 0.2s ease-in-out;
  }

  .navigation__menuButton {
    display: block;
    width: 48px;
    height: 48px;
    cursor: pointer;
    z-index: 9999;
    border-radius: 50%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  .navigation__menuButtonLine {
    display: block;
    width: 20px;
    height: 2px;
    background: #ddd;
    border-radius: 2px;
    margin: 1px;
  }
}

.navigation__input:checked + .navigation__menuButton {
  transform: rotate(45deg);
}

.navigation__input:checked + .navigation__menuButton i {
  background: #000;
  transition: transform 0.2s ease-in-out;
}

.navigation__input:checked + .navigation__menuButton i:nth-child(1) {
  transform: rotate(180deg);
}

.navigation__input:checked + .navigation__menuButton i:nth-child(2) {
  opacity: 0;
}

.navigation__input:checked + .navigation__menuButton i:nth-child(3) {
  transform: translateY(-8px) rotate(90deg);
}

.navigation__input:checked ~ .navigation__listWrapper {
  z-index: 9990;
  opacity: 1;
}

.navigation__input:checked ~ .navigation__listWrapper ul {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
}

.navigation__input:checked ~ .navigation__listWrapper ul li a {
  opacity: 1;
  transform: translateX(0);
}

.navigation__input {
  display: none;
}
