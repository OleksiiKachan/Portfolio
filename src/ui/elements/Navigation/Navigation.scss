@import '../../assets/style/variables';
@import '../../assets//style/keyframes';
@import '../../assets//style/mixins';

.navigation {
  padding: 8px;
  height: 15%;

  background-color: $color-surface;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  transition: all 0.4s ease-in-out;

  position: relative;
}

.navigation__logoWrapper {
  padding: 16px;
  width: 35%;
}

.navigation__logoImage {
  width: 40%;
}

.navigation__menuButton {
  width: 0;
}

.navigation__logoWrapper {
  transition: all 0.4s ease-in-out;
}

.navigation__logoImage {
  transition: all 0.4s ease-in-out;
}

.navigation__listWrapper {
  padding: 0 34px;
  width: 100%;
  transition: all 0.4s ease-in-out;

  display: flex;
  justify-content: center;
  align-items: center;
}

.navigation__list {
  list-style: none;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  width: 100%;
}

.navigation__listLink {
  display: inline-block;
  outline: none;
  color: $color-black;
  text-decoration: none;
  font-size: 14px;
  letter-spacing: 1.2px;
  font-weight: 500;
}

@media screen and (min-width: $breakpoint-tablet) {
  .navigation__listItem {
    &_selected {
      &:after {
        animation: widthGrow 300ms ease-in-out forwards;
      }
      .navigation__listLink {
        font-weight: 800;
      }
    }

    &:after {
      content: '';
      background-color: $color-black;
      display: block;
      width: 0;
      height: 1px;
      margin-top: 8px;
    }

    &:hover {
      &:after {
        animation: widthGrow 300ms ease-in-out forwards;
      }
    }
  }
}

@media screen and (max-width: 864px) {
  .navigation__logoWrapper {
    padding: 0 8px;
    display: flex;
    justify-content: flex-start;
    align-items: center;
  }
  .navigation__logoWrapper {
    width: 25%;
  }

  .navigation__logoImage {
    content: url('https://res.cloudinary.com/oleksiikachan/image/upload/v1546142703/portfolio/images/logo/logo_square_dark.svg');
    width: 45%;
  }

  .navigation__listWrapper {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: -1;
    background: $color-surface;
    opacity: 0;
    transition: display 0.5s ease-in-out;
  }

  .navigation__list {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
  }

  .navigation__listItem {
    display: block;
    float: none;
    width: 100%;
    text-align: right;
    margin-bottom: 10px;
  }

  .navigation__list li:nth-child(1) a {
    transition-delay: 0.2s;
  }

  .navigation__list li:nth-child(2) a {
    transition-delay: 0.3s;
  }

  .navigation__list li:nth-child(3) a {
    transition-delay: 0.4s;
  }

  .navigation__list li:nth-child(4) a {
    transition-delay: 0.5s;
  }

  .navigation__list li:nth-child(5) a {
    transition-delay: 0.6s;
  }

  .navigation__listLink {
    padding: 10px 24px;
    opacity: 0;
    color: $color-grey;
    font-size: 14px;
    font-weight: 600;
    letter-spacing: 1.2px;
    transform: translateX(-20px);
    transition: all 0.2s ease-in-out;
  }

  .navigation__menuButton {
    display: block;
    width: 48px;
    height: 48px;
    cursor: pointer;
    z-index: 9999;
    border-radius: 50%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }

  .navigation__menuButtonLine {
    display: block;
    width: 20px;
    height: 2px;
    background: $color-black;
    border-radius: 2px;
    margin: 1px;
  }
}

.navigation__input:checked + .navigation__menuButton {
  transform: rotate(45deg);
}

.navigation__input:checked + .navigation__menuButton i {
  background: $color-black;
  transition: transform 0.2s ease-in-out;
}

.navigation__input:checked + .navigation__menuButton i:nth-child(1) {
  transform: rotate(180deg);
}

.navigation__input:checked + .navigation__menuButton i:nth-child(2) {
  opacity: 0;
}

.navigation__input:checked + .navigation__menuButton i:nth-child(3) {
  transform: translateY(-8px) rotate(90deg);
}

.navigation__input:checked ~ .navigation__listWrapper {
  z-index: 9990;
  opacity: 1;
}

.navigation__input:checked ~ .navigation__listWrapper ul {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
}

.navigation__input:checked ~ .navigation__listWrapper ul li a {
  opacity: 1;
  transform: translateX(0);
}

.navigation__input {
  display: none;
}
