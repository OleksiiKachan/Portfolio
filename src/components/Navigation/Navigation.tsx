import React, { Component } from 'react';
import classnames from 'classnames';
import './Navigation.scss';
import NavigationLogo from '../NavigationLogo';
import MenuIconToggle from '../MenuIconToggle';
import NavigationList from '../NavigationList';

/*
    <Navigation navigationItems=[]/>
*/

type PropsType = {
  navigationItems: Array<{
    key: string;
    title: string;
    sectionUrl: string;
    isSelected: boolean;
  }>;
  onLinkClick: (selectedItemKey: string) => void;
  className?: string;
};

class Navigation extends Component<PropsType> {
  render() {
    return (
      <nav className={classnames('header__navigation', 'navigation')}>
        <NavigationLogo />
        <MenuIconToggle />
        <NavigationList
          onLinkClick={this.props.onLinkClick}
          navigationItems={this.props.navigationItems}
        />
      </nav>
    );
  }
}

export default Navigation;
