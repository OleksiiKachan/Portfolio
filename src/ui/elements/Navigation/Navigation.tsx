import React, { Component } from 'react';
import classnames from 'classnames';
import { NavigationItemType } from '../../../lib/types';
import NavigationLogo from './NavigationLogo/NavigationLogo';
import MenuIconToggle from './MenuIconToggle/MenuIconToggle';
import NavigationList from './NavigationList/NavigationList';
import './Navigation.scss';

/*
    <Navigation
      navigationItems=[]
      onLinkClick=() => {}
      className=''
    />
*/

type PropsType = {
  navigationItems: Array<NavigationItemType>;
  onLinkClick: (selectedItemKey: string) => void;
  className?: string;
};

export default class Navigation extends Component<PropsType> {
  render() {
    return (
      <nav className={classnames('header__navigation', 'navigation')}>
        <NavigationLogo />
        <MenuIconToggle />
        <NavigationList
          onLinkClick={this.props.onLinkClick}
          navigationItems={this.props.navigationItems}
        />
      </nav>
    );
  }
}
