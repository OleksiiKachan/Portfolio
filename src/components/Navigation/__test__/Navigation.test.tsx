import React from 'react';
import { shallow } from 'enzyme';
import Navigation from '..';

describe('Navigation', () => {
  it('Snapshot', () => {
    const component = shallow(
      <Navigation
        navigationItems={[
          {
            key: 'mi_education',
            title: 'Education & Experience',
            sectionUrl: '/experience',
            isSelected: false,
          },
          {
            key: 'mi_contact',
            title: 'Contact',
            sectionUrl: '/contact',
            isSelected: false,
          },
        ]}
        onLinkClick={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
