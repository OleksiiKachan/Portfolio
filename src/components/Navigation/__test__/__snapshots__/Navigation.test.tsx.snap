// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Snapshot 1`] = `
<nav
  className="header__navigation navigation"
>
  <_default />
  <_default />
  <NavigationList
    navigationItems={
      Array [
        Object {
          "isSelected": false,
          "key": "mi_education",
          "sectionUrl": "/experience",
          "title": "Education & Experience",
        },
        Object {
          "isSelected": false,
          "key": "mi_contact",
          "sectionUrl": "/contact",
          "title": "Contact",
        },
      ]
    }
    onLinkClick={[Function]}
  />
</nav>
`;
